### ISSUE TITLE
Cannot compile with boost 1.79

### ISSUE THREAD

**Issue by intractabilis** [kallaballa/libnfporb#29]
There is some conflict between boost::qvm and LongDouble:
```
make -C examples/  CXX=g++ NVCC="" NVCC_HOST_CXX="" NVCC_CXXFLAGS=""
make[1]: Entering directory '/home/user/libs-dev/libnfp/examples'
g++ -pthread -fno-strict-aliasing -std=c++1y -pedantic -Wall -march=native -g0 -O3 -fpic -I. -o nfp.o -c nfp.cpp
In file included from /home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/strategies.hpp:117,
                 from /home/user/libs-dev/boost_1_79_0/boost/geometry/geometry.hpp:60,
                 from /home/user/libs-dev/boost_1_79_0/boost/geometry.hpp:17,
                 from ../src/geometry.hpp:13,
                 from ../src/libnfporb.hpp:12,
                 from nfp.cpp:7:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp: In instantiation of ‘boost::geometry::strategy::transform::matrix_transform
er<CalculationType, 2, 2>::matrix_transformer(const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&) [with CalculationType = libnf
porb::LongDouble; ct = libnfporb::LongDouble]’:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:336:24:   required from ‘boost::geometry::strategy::transform::translate_tra
nsformer<CalculationType, 2, 2>::translate_transformer(const CalculationType&, const CalculationType&, const CalculationType&) [with CalculationType = libnfporb::LongDouble]’
../src/algo/trim_vector.hpp:22:87:   required from here
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:185:20: error: no matching function for call to ‘A<0, 0>(boost::geometry::st
rategy::transform::detail::matrix_transformer::matrix_transformer<libnfporb::LongDouble, 2, 2>::matrix_type&)’
  185 |         qvm::A<0,0>(this->m_matrix) = m_0_0;   qvm::A<0,1>(this->m_matrix) = m_0_1;   qvm::A<0,2>(this->m_matrix) = m_0_2;
      |         ~~~~~~~~~~~^~~~~~~~~~~~~~~~
In file included from /home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:27:
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp:21:1: note: candidate: ‘template<int R, int C, class M> typename boost::qvm::enable_if_c<boost::qvm::is_mat<M>:
:value, typename boost::qvm::mat_traits<M>::scalar_type>::type boost::qvm::A(const M&)’
   21 | A( M const & a )
      | ^
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp:21:1: note:   template argument deduction/substitution failed:
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp: In substitution of ‘template<int R, int C, class M> typename boost::qvm::enable_if_c<boost::qvm::is_mat<M>::va
lue, typename boost::qvm::mat_traits<M>::scalar_type>::type boost::qvm::A(const M&) [with int R = 0; int C = 0; M = boost::qvm::mat<libnfporb::LongDouble, 3, 3>]’:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:185:20:   required from ‘boost::geometry::strategy::transform::matrix_transf
ormer<CalculationType, 2, 2>::matrix_transformer(const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&) [with CalculationType = li
bnfporb::LongDouble; ct = libnfporb::LongDouble]’
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:336:24:   required from ‘boost::geometry::strategy::transform::translate_tra
nsformer<CalculationType, 2, 2>::translate_transformer(const CalculationType&, const CalculationType&, const CalculationType&) [with CalculationType = libnfporb::LongDouble]’
../src/algo/trim_vector.hpp:22:87:   required from here
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp:21:1: error: no type named ‘type’ in ‘struct boost::qvm::enable_if_c<false, libnfporb::LongDouble>’
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp: In instantiation of ‘boost::geometry::strategy::transform::matrix_transform
er<CalculationType, 2, 2>::matrix_transformer(const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&) [with CalculationType = libnf
porb::LongDouble; ct = libnfporb::LongDouble]’:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:336:24:   required from ‘boost::geometry::strategy::transform::translate_tra
nsformer<CalculationType, 2, 2>::translate_transformer(const CalculationType&, const CalculationType&, const CalculationType&) [with CalculationType = libnfporb::LongDouble]’
../src/algo/trim_vector.hpp:22:87:   required from here
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp:35:1: note: candidate: ‘template<int R, int C, class M> typename boost::qvm::enable_if_c<boost::qvm::is_mat<M>:
:value, typename boost::qvm::mat_traits<M>::scalar_type&>::type boost::qvm::A(M&)’
   35 | A( M & a )
      | ^
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp:35:1: note:   template argument deduction/substitution failed:
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp: In substitution of ‘template<int R, int C, class M> typename boost::qvm::enable_if_c<boost::qvm::is_mat<M>::va
lue, typename boost::qvm::mat_traits<M>::scalar_type&>::type boost::qvm::A(M&) [with int R = 0; int C = 0; M = boost::qvm::mat<libnfporb::LongDouble, 3, 3>]’:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:185:20:   required from ‘boost::geometry::strategy::transform::matrix_transf
ormer<CalculationType, 2, 2>::matrix_transformer(const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&) [with CalculationType = li
bnfporb::LongDouble; ct = libnfporb::LongDouble]’
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:336:24:   required from ‘boost::geometry::strategy::transform::translate_tra
nsformer<CalculationType, 2, 2>::translate_transformer(const CalculationType&, const CalculationType&, const CalculationType&) [with CalculationType = libnfporb::LongDouble]’
../src/algo/trim_vector.hpp:22:87:   required from here
/home/user/libs-dev/boost_1_79_0/boost/qvm/mat_access.hpp:35:1: error: no type named ‘type’ in ‘struct boost::qvm::enable_if_c<false, libnfporb::LongDouble&>’
In file included from /home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:28:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp: In instantiation of ‘boost::geometry::strategy::transform::matrix_transform
er<CalculationType, 2, 2>::matrix_transformer(const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&, const ct&) [with CalculationType = libnf
porb::LongDouble; ct = libnfporb::LongDouble]’:
/home/user/libs-dev/boost_1_79_0/boost/geometry/strategies/transform/matrix_transformers.hpp:336:24:   required from ‘boost::geometry::strategy::transform::translate_tra
nsformer<CalculationType, 2, 2>::translate_transformer(const CalculationType&, const CalculationType&, const CalculationType&) [with CalculationType = libnfporb::LongDouble]’
../src/algo/trim_vector.hpp:22:87:   required from here
/home/user/libs-dev/boost_1_79_0/boost/qvm/vec_access.hpp:21:1: note: candidate: ‘template<int I, class V> typename boost::qvm::enable_if_c<boost::qvm::is_vec<B>::value,
 typename boost::qvm::vec_traits<B>::scalar_type>::type boost::qvm::A(const V&)’
   21 | A( V const & a )
      | ^
/home/user/libs-dev/boost_1_79_0/boost/qvm/vec_access.hpp:21:1: note:   template argument deduction/substitution failed:
...
```

**Comment by intractabilis** [comment]
I had to add
```
namespace boost { namespace qvm {
    template <> struct is_scalar<libnfporb::LongDouble> { static bool const value = true; };
}}
```
at the end of geometry.hpp to fix this.

**Comment by kallaballa** [comment]
Oh cool! Thx for the fix, I'll add and test it!

**Comment by kallaballa** [comment]
fixed 0685e55e73ca14e47722a738f4ccd9bd6d9419a7
